### PR TITLE
Bug 1970332: Fix page disappearing issue while creating sc for rbd provisioner

### PR DESCRIPTION
**Pull request by SanjalKatiyar**

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx [hunk 1]
```diff
@@ -63,7 +63,8 @@ export const KMSConfigure: React.FC<KMSConfigureProps> = ({ state, dispatch, mod
 
   const setAddressPort = (port: string) => {
     kmsObj.port.value = port;
-    kmsObj.port.valid = port !== '' && !_.isNaN(Number(port)) && Number(port) > 0;
+    kmsObj.port.valid =
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx:66]: Is patternfly is providing any port component?

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx:66]: I checked patternfly only supports following types for `<TextInput>`:
```
"number" | "time" | "text" | "url" | "date" | "datetime-local" | "email" | "month" | "password" | "search" | "tel"
```

File: frontend/public/components/storage-class-form.tsx [hunk 2]
```diff
@@ -708,7 +708,7 @@ class StorageClassFormWithTranslation extends React.Component<
     const dataParameters = _.pickBy(
       _.mapValues(this.state.newStorageClass.parameters, (value, key) => {
         let finalValue = value.value;
-        if (this.storageTypes[type].parameters[key].format) {
+        if (this.storageTypes[type].parameters[key]?.format) {
```
Comment (SanjalKatiyar) [inline frontend/public/components/storage-class-form.tsx:711]: Exact steps are little harder to reproduce, but, this line was causing `Cannot read property 'format' of undefined` error for certain cases.
Anyways, it is safer to use `?.` than not adding it at all.

Comment (bipuladh) [reply frontend/public/components/storage-class-form.tsx:711]: @rawagner PTAL

**Pull request merged**